scripts/postinstall.js

```diff
@@ -0,0 +1,197 @@
+}
+
+/**
+ * Prompt the user through the configurable firebase add-on services
+ */
+function promptQuestions() {
+    prompt.get([{
+        name: 'remote_config',
+        description: 'Are you using Firebase RemoteConfig (y/n)',
+        default: 'n',
+    }, {
+        name: 'messaging',
+        description: 'Are you using Firebase Messaging (y/n)',
+        default: 'n'
+    }, {
+        name: 'storage',
+        description: 'Are you using Firebase Storage (y/n)',
+        default: 'n'
+    }, {
+        name: 'facebook_auth',
+        description: 'Are you using Facebook Authentication (y/n)',
+        default: 'n'
+    }, {
+        name: 'google_auth',
+        description: 'Are you using Google Authentication (y/n)',
+        default: 'n'
+    }], function (err, result) {
+        if (err) {
+            return console.log(err);
+        }
+        if(usingiOS) {
+            writePodFile(result);
+        }
+        if(usingAndroid) {
+            writeGradleFile(result);
+        }
+    });
+}
+
+/**
+ * Create the iOS PodFile for installing the Firebase iOS dependencies and service dependencies
+ *
+ * @param {any} result The answers to the micro-service prompts
+ */
+function writePodFile(result) {
+    if(!fs.existsSync(directories.ios)) {
+        fs.mkdirSync(directories.ios);
+    }
+    fs.writeFile(directories.ios + '/Podfile',
+    `pod 'Firebase', '~> 3.3.0'
+    pod 'Firebase/Database'
+    pod 'Firebase/Auth'
+    pod 'Firebase/Crash'
+
+    # Uncomment if you want to enable Remote Config
+    ` + (isSelected(result.remote_config) ? `` : `#`) + `pod 'Firebase/RemoteConfig'
+
+    # Uncomment if you want to enable FCM (Firebase Cloud Messaging)
+    ` + (isSelected(result.messaging) ? `` : `#`) + `pod 'Firebase/Messaging'
+
+    # Uncomment if you want to enable Firebase Storage
+    ` + (isSelected(result.storage) ? `` : `#`) + `pod 'Firebase/Storage'
+
+    # Uncomment if you want to enable Facebook Authentication
+    ` + (isSelected(result.facebook_auth) ? `` : `#`) + `pod 'FBSDKCoreKit'
+    ` + (isSelected(result.facebook_auth) ? `` : `#`) + `pod 'FBSDKLoginKit'
+
+    # Uncomment if you want to enable Google Authentication
+    ` + (isSelected(result.google_auth) ? `` : `#`) + `pod 'GoogleSignIn'`, function(err) {
+        if(err) {
+            return console.log(err);
+        }
+        console.log('Successfully created iOS (Pod) file.');
+    });
+}
+
+/**
+ * Create the Android Gradle for installing the Firebase Android dependencies and service dependencies
+ *
+ * @param {any} result The answers to the micro-service prompts
+ */
+function writeGradleFile(result) {
+     if(!fs.existsSync(directories.android)) {
+        fs.mkdirSync(directories.android);
+    }
+    fs.writeFile(directories.android + '/include.gradle',
+    `
+    android {
+        productFlavors {
+            "fireb" {
+                dimension "fireb"
+            }
+        }
+    }
+
+    repositories {
+        mavenCentral()
```

Comment:
With how gradle works, gradle selects the first repository that contains the required dependency and uses that.  per [this release](http://www.technotalkative.com/android-studio-migration-maven-central-jcenter/) Android Studio migrated their default repository to jcenter.

jcenter is a superset of mavenCentral with the added bonus of serving their content over HTTPS and a CDN! 

My recommendation would be to place jcenter first so at least the repositories will default to this preference. The more extreme being that since it's a superset, removing mavenCentral shouldn't have any issues.